mne_bids/meg_bids.py

```diff
@@ -267,99 +247,114 @@ def raw_to_bids(subject_id, session_id, run, task, raw_fname, output_path,
-
-    # CTF systems
-    elif ext == '.ds':
-        raw = io.read_raw_ctf(raw_fname)
+    if isinstance(raw_file, str):
+        # We must read in the raw data
+        raw = _read_raw(raw_file, electrode=electrode, hsp=hsp, hpi=hpi,
+                        config=config, verbose=verbose)
+        _, ext = _parse_ext(raw_file)
+    elif isinstance(raw_file, BaseRaw):
+        # Only parse the filename for the extension
+        # Assume that if no filename attr exists, it's a fif file.
+        raw = raw_file
+        if hasattr(raw, 'filenames'):
+            _, ext = _parse_ext(raw.filenames[0])
+        else:
+            ext = '.fif'
+    data_path = make_bids_folders(subject=subject_id, session=session_id,
+                                  kind=kind, root=output_path,
+                                  overwrite=overwrite,
+                                  verbose=verbose)
+    if session_id is None:
+        ses_path = data_path
+    else:
+        ses_path = make_bids_folders(subject=subject_id, session=session_id,
+                                     root=output_path,
+                                     overwrite=False,
+                                     verbose=verbose)
+
+    # create filenames
+    scans_fname = make_bids_filename(
+        subject=subject_id, session=session_id, suffix='scans.tsv',
+        prefix=ses_path)
+
+    coordsystem_fname = make_bids_filename(
+        subject=subject_id, session=session_id,
+        suffix='coordsystem.json', prefix=data_path)
+    data_meta_fname = make_bids_filename(
+        subject=subject_id, session=session_id,
+        suffix='%s.json' % kind, prefix=data_path)
+    raw_file_bids = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        suffix='%s%s' % (kind, ext), prefix=data_path)
+    events_tsv_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task,
+        run=run, suffix='events.tsv', prefix=data_path)
+    channels_fname = make_bids_filename(
+        subject=subject_id, session=session_id, task=task, run=run,
+        suffix='channels.tsv', prefix=data_path)
+
+    # Read in Raw object and extract metadata from Raw object if needed
+    if kind == 'meg':
+        orient = orientation[ext]
+        unit = units[ext]
+        manufacturer = manufacturers[ext]
+    else:
+        orient = None
+        unit = None
+        manufacturer = None
 
     # save stuff
-    _dataset_description_json(output_path, verbose)
-    _scans_tsv(raw, raw_fname_bids, scans_fname, verbose)
-    _coordsystem_json(raw, unit, orient, manufacturer, fid_fname, verbose)
-    _meg_json(raw, task, manufacturer, meg_fname, verbose)
+    if kind == 'meg':
+        _scans_tsv(raw, raw_file_bids, scans_fname, verbose)
+        _coordsystem_json(raw, unit, orient, manufacturer, coordsystem_fname,
+                          verbose)
+
+    make_dataset_description(output_path, name=" ",
```

Comment:
why is this public? I think we should really be mindful of what we make public

Reply:
In this case, because I saw that people in the `bids-examples` folder were including example scripts etc to create this file, and thought that it's a common-enough operation that we may as well create a helper function for it. In my mind, you could use MNE-BIDS to do the initial raw conversion, and then the `utils` functions to do any tweaking that's needed

Reply:
hmm ... but there is only one `dataset_description` in the folder. You don't think it's easier to edit the  text file than write a function to do it for you? :-P

Reply:
I think we should just put `n/a` whatever we can't fill in. Then the user can simply grep for `n/a` and fill up those fields. 

Maybe you would need functions to do that. But let's not overfit. For this PR, we should focus on just iEEG and separating raw i/o from folder creation. One step at a time :-)

Reply:
that works too - though I anticipate using mne-bids to convert data that's not in `.fif` format, so in my case I'll know about the hidden function but other users in a similar situation (e.g. anybody with ieeg data most likely) won't know about this

Reply:
re: n/a, I think this should be solved a different way. some of the fields actually can't have strings, (e.g. they expect arrays) so we'd need to hard-code that in there. I think instead we can prototype something like having JSON templates of the proper form in a subsequent PR.